src/v/features/migrators/cloud_storage_config.cc

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "features/migrators/cloud_storage_config.h"
+
+#include "cluster/controller.h"
+#include "cluster/feature_manager.h"
+#include "cluster/topics_frontend.h"
+#include "features/feature_migrator.h"
+#include "features/feature_table.h"
+#include "features/logger.h"
+#include "ssx/future-util.h"
+#include "vlog.h"
+
+#include <seastar/core/sleep.hh>
+
+#include <type_traits>
+
+using namespace std::chrono_literals;
+
+namespace features::migrators {
+
+template<typename T>
+cluster::property_update<T> make_property_set(const T& v) {
+    return cluster::property_update<T>(
+      v, cluster::incremental_update_operation::set);
+}
+
+ss::future<> cloud_storage_config::do_mutate() {
+    cluster::topic_table& topic_table = _controller.get_topics_state().local();
+
+    vlog(featureslog.info, "Checking for topic configs to update...");
+    size_t update_count = 0;
+    for (const auto& i : topic_table.topics_map()) {
+        auto& topic_conf = i.second.metadata.get_configuration();
+
+        auto& props = topic_conf.properties;
+
+        bool si_enabled
+          = (props.shadow_indexing
+             && props.shadow_indexing
+                  != model::shadow_indexing_mode::disabled)
+            || ((!props.shadow_indexing || props.shadow_indexing == model::shadow_indexing_mode::disabled) && config::shard_local_cfg().cloud_storage_enable_remote_write());
```

Comment:
This condition enlarges the set of topics that we apply the migration to (compared to what we had previously). It includes topics that are only enabled for remote *read* via the topic level config. Was this intentional? Also, the handling of remote *read* only topics is not consistent. It depends on whether it was enabled through the cluster or topic config.

Reply:
It wasn't exactly intentional, I was kind of ignoring read replicas because they retention settings aren't meaningful.  However for hygiene it's better not the gratuitously change the rules, so I've modified to use the same condition that was in maybe_adjust_retention_policies before.